src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -578,9 +582,23 @@ export default class LazyMobXTable<T> extends React.Component<LazyMobXTableProps
         showCountHeader: false
     };
 
-    public getDownloadData(): string
+    public getDownloadData(): Promise<string>
     {
-        return serializeData(this.store.downloadData);
+        return new Promise<string>((resolve, reject) => {
+            if (this.store.downloadStore) {
+                this.store.downloadStore.loadAllData().then(isLoaded => {
+                    if (isLoaded) {
+                        resolve(serializeData(this.store.downloadData));
```

Comment:
I think the use of this is confusing, where the downloadstore only reports on the data status and requests it but does not return it. I think the downloadstore should return the data as well